File: lib/antlr-2.7.5/examples/java/exprAST/Main.java
import java.io.*;
import antlr.collections.AST;

class Main {
	public static void main(String[] args) {
		try {
			ExprLexer lexer = new ExprLexer(new DataInputStream(System.in));
			ExprParser parser = new ExprParser(lexer);

			// set the type of tree node to create; this is default action
			// so it is unnecessary to do it here, but demos capability.
			parser.setASTNodeType("antlr.CommonAST");

			parser.expr();
			antlr.CommonAST ast = (antlr.CommonAST)parser.getAST();
			if (ast != null) {
				System.out.println(ast.toStringList());
			} else {
				System.out.println("null AST");
			}
		} catch(Exception e) {
			System.err.println("exception: "+e);
		}
	}
}

File: lib/antlr-2.7.5/examples/cpp/exprAST/Main.cpp

#include <iostream>

#include "ExprLexer.hpp"
#include "ExprParser.hpp"
#include "antlr/AST.hpp"
#include "antlr/CommonAST.hpp"

int main( int, char** )
{
	ANTLR_USING_NAMESPACE(std)
	ANTLR_USING_NAMESPACE(antlr)
	try {
		ASTFactory my_factory;
		ExprLexer lexer(cin);
		ExprParser parser(lexer);

		parser.initializeASTFactory(my_factory);
		parser.setASTFactory(&my_factory);

		parser.expr();
		RefCommonAST ast = RefCommonAST(parser.getAST());

		if (ast)
			cout << ast->toStringList() << endl;
		else
			cout << "null AST" << endl;
	}
	catch( ANTLRException& e )
	{
		cerr << "exception: " << e.getMessage() << endl;
		return -1;
	}
	catch( exception& e )
	{
		cerr << "exception: " << e.what() << endl;
		return -1;
	}
	return 0;
}

